apps/etterlatte-utbetaling/src/main/kotlin/no/nav/etterlatte/utbetaling/avstemming/regulering/VerifiserReguleringssummer.kt

```diff
@@ -0,0 +1,97 @@
+                logger.warn("Klarte ikke å verifisere match mellom utbetaling og vedtak for vedtak $it", e)
+                logger.debug("Fortsetter med å verifisere neste vedtak")
+            }
+        }
+    }
+
+    suspend fun verifiser(vedtakId: Long) {
+        val utbetaling = repository.hentUtbetaling(vedtakId = vedtakId)
+        if (utbetaling == null) {
+            logger.warn("Ingen utbetaling for vedtak $vedtakId. Returnerer")
+            return
+        }
+        val vedtak =
+            retryOgPakkUt {
+                vedtaksvurderingKlient.hentVedtak(
+                    utbetaling.behandlingId.value,
+                    Systembruker.automatiskJobb,
+                )
+            }
+        sammenlignLinjer(utbetaling, vedtak)
+    }
+
+    private fun sammenlignLinjer(
+        utbetaling: Utbetaling,
+        vedtak: VedtakDto,
+    ) {
+        if (vedtak.innhold is VedtakInnholdDto.VedtakBehandlingDto) {
+            (vedtak.innhold as VedtakInnholdDto.VedtakBehandlingDto).utbetalingsperioder.forEach { fraVedtak ->
+                val korresponderendeUtbetalingslinje =
+                    utbetaling.utbetalingslinjer
+                        .maxByOrNull { YearMonth.from(it.periode.fra) == fraVedtak.periode.fom }
+                        ?: throw IllegalStateException(
+                            "Mangler korresponderende utbetalingslinje for periode ${fraVedtak.periode} for vedtak ${vedtak.id}",
+                        )
+                verifiserAtStemmerOverens(vedtak.id, fraVedtak, korresponderendeUtbetalingslinje)
+            }
+        }
+    }
+
+    private fun verifiserAtStemmerOverens(
+        vedtakId: Long,
+        fraVedtak: Utbetalingsperiode,
+        korresponderendeUtbetalingslinje: Utbetalingslinje,
+    ) {
+        check(sammenlignBeloep(fraVedtak, korresponderendeUtbetalingslinje)) {
+            "Beløp fra vedtak $vedtakId var ${fraVedtak.beloep}, men i utbetalingslinje ${korresponderendeUtbetalingslinje.beloep}"
+        }
+        check(fraVedtak.periode.tom == korresponderendeUtbetalingslinje.periode.til?.let { YearMonth.from(it) }) {
+            "Tom-periode fra vedtak $vedtakId var ${fraVedtak.periode.tom}, men i utbetalingslinje ${
+                korresponderendeUtbetalingslinje.periode.til?.let {
+                    YearMonth.from(
+                        it,
+                    )
+                }
+            }"
+        }
+    }
+
+    private fun sammenlignBeloep(
+        fraVedtak: Utbetalingsperiode,
+        korresponderendeUtbetalingslinje: Utbetalingslinje,
+    ): Boolean {
+        if (fraVedtak.beloep == null && korresponderendeUtbetalingslinje.beloep == null) {
+            return true
+        }
+        if (fraVedtak.beloep == null || korresponderendeUtbetalingslinje.beloep == null) {
+            return false
+        }
+        return fraVedtak.beloep!!.minus(korresponderendeUtbetalingslinje.beloep).abs() < BigDecimal.ONE
```

Comment:
Kan dette noen gang være annet enn 0 da vi opererer med heltall i vedtakene våre?

Reply:
Bør ikkje vera det, men datatypen i basen er numeric, utan presisert scale, så da _If you omit precision and scale, they will default to 131072 and 16383, respectively._ Og vi les det inn igjen som bigdecimal.

Så konklusjon: er usikker.